### PR TITLE
[fix](cdc)fix mongodb sync cause ClassCastException when user specifies the `_id` field manually

### PR DESCRIPTION
**Pull request by qg-lin**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

When user specifies the `_id` field manually by Long type or other basic types, and forced cast it to Map type by code  `((Map<?, ?>) rowMap.get(ID_FIELD))` would ClassCastException.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by qg-lin** [comment]
@JNSimba PTAL

File: flink-doris-connector/src/main/java/org/apache/doris/flink/tools/cdc/CdcTools.java [hunk 1]
```diff
@@ -161,7 +161,10 @@ private static void syncDatabase(
             jobName =
                     String.format(
                             "%s-Doris Sync Database: %s",
-                            type, config.getString("database-name", "db"));
+                            type,
+                            "mongodb".equals(type)
+                                    ? config.getString("database", "db")
+                                    : config.getString("database-name", "db"));
```
Comment (bingquanzhao) [inline flink-doris-connector/src/main/java/org/apache/doris/flink/tools/cdc/CdcTools.java:167]: It is better to unify it as database-name. You can change the config in MongoDBDatabaseSync.buildCdcSource to read

Comment (qg-lin) [reply flink-doris-connector/src/main/java/org/apache/doris/flink/tools/cdc/CdcTools.java]: thanks for you reviewing, another commit changed it.

**Review by JNSimba — APPROVED** [review]
LGTM

**Pull request merged**